### PR TITLE
2️⃣ Use v2 api for persisting sessions

### LINKED ISSUE #20637: 2️⃣ Use v2 api for persisting Wallet Connect sessions
# Feature Issue

### User Story

As a developer, I want to persist sessions using `wallet_addWalletConnectSession` api **so that** we can handle WC v2 and are at parity with Desktop. 

### Description
While working with persistence I (@shivekkhurana) integrated the v1 api, following Desktop as a reference. V1 api exposes the method `wakuext_addWalletConnectSession` and v2 api exposes `wallet_addWalletConnectSession`.

### Acceptance Criteria
- [ ] Sessions should be persisted using the v2 api

### PR DESCRIPTION
**Pull request by shivekkhurana**
Fixes #20637

### CODE REVIEW

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 1]
```diff
@@ -213,29 +206,32 @@
...
+   {:fx [[:json-rpc/call
+          [{:method     "wallet_addWalletConnectSession"
+            :params     [(js/JSON.stringify session-info)]
+            :on-success #(log/info "Wallet Connect session persisted")
+            :on-error   #(log/info "Wallet Connect session persistence failed" %)}]]]}))
+
+(comment
+  (rf/dispatch [:wallet-connect/fetch-persisted-sessions-v2])
+  ;;
+)
```
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs]: can be removed

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 2]
```diff
@@ -213,29 +206,32 @@
...
-            :on-error   #(log/info "Wallet Connect session persistence failed" %)}]]]}))
-
 (rf/reg-event-fx
  :wallet-connect/fetch-persisted-sessions-success
  (fn [{:keys [db]} [sessions]]
    {:db (assoc db :wallet-connect/persisted-sessions sessions)}))
 
 (rf/reg-event-fx
- :wallet-connect/fetch-persisted-sessions
+ :wallet-connect/fetch-persisted-sessions-v2
```
Comment (clauxx) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs]: don't think we need the v2 here

**Review by clauxx — APPROVED** [review]

**Review by alwx — APPROVED** [review]

**Comment by clauxx** [comment]
@pavloburykh are the failed tests related or is it ok to merge?

**Comment by qoqobolo** [comment]
@shivekkhurana @clauxx the e2e failures here are also not related to the PR.
It can be merged.

**Pull request merged**